kmap/resources.go

```diff
@@ -0,0 +1,388 @@
+
+		for _, service := range services.Items {
+			var blankResource Resource
+			blankResource.name = service.Name
+			blankResource.kind = "Service"
+			blankResource.Labels = service.Labels
+			blankResource.selector = service.Spec.Selector
+			rPool.addToResourcePool(&blankResource)
+		}
+	}
+	return rPool
+}
+func addServiceAccounts(serviceAccs *v1.ServiceAccountList, rPool *ResourcePool) *ResourcePool {
+	if len(serviceAccs.Items) > 0 {
+
+		for _, serviceAcc := range serviceAccs.Items {
+			var blankResource Resource
+			blankResource.name = serviceAcc.Name
+			blankResource.kind = "ServiceAccount"
+			blankResource.Labels = serviceAcc.Labels
+			rPool.addToResourcePool(&blankResource)
+		}
+
+	}
+	return rPool
+}
+func addDaemonSets(daemonsets *appsv1.DaemonSetList, rPool *ResourcePool) *ResourcePool {
+	if len(daemonsets.Items) > 0 {
+
+		for _, ds := range daemonsets.Items {
+			var blankResource Resource
+			blankResource.name = ds.Name
+			blankResource.kind = "DaemonSet"
+			blankResource.status = string(ds.Status.NumberReady) + "/" + string(ds.Status.DesiredNumberScheduled)
+			blankResource.Labels = ds.Labels
+			blankResource.selector = ds.Spec.Selector.MatchLabels
+			rPool.addToResourcePool(&blankResource)
+		}
+
+	}
+	return rPool
+}
+func addDeployments(deployments *appsv1.DeploymentList, rPool *ResourcePool) *ResourcePool {
+	if len(deployments.Items) > 0 {
+		for _, deployment := range deployments.Items {
+			var blankResource Resource
+			blankResource.name = deployment.Name
+			blankResource.kind = "Deployment"
+			var readyReplica int32 = deployment.Status.ReadyReplicas
+			var availableReplica int32 = deployment.Status.AvailableReplicas
+			blankResource.status = Int32toString(readyReplica) + "/" + Int32toString(availableReplica)
+			blankResource.Labels = deployment.Labels
+			blankResource.selector = deployment.Spec.Selector.MatchLabels
+			rPool.addToResourcePool(&blankResource)
+		}
+	}
+	return rPool
+}
+func addReplicaSets(rsets *appsv1.ReplicaSetList, rPool *ResourcePool) *ResourcePool {
+	if len(rsets.Items) > 0 {
+		for _, rs := range rsets.Items {
+			var blankResource Resource
+			blankResource.name = rs.Name
+			blankResource.kind = "Replicaset"
+			blankResource.Labels = rs.Labels
+			blankResource.selector = rs.Spec.Selector.MatchLabels
+			rPool.addToResourcePool(&blankResource)
+		}
+	}
+	return rPool
+}
+func addStateFulSets(ssets *appsv1.StatefulSetList, rPool *ResourcePool) *ResourcePool {
+	if len(ssets.Items) > 0 {
+		for _, sset := range ssets.Items {
+			var blankResource Resource
+			blankResource.name = sset.Name
+			blankResource.kind = "StatefulSet"
+			blankResource.status = Int32toString(sset.Status.ReadyReplicas) + "/" + Int32toString(sset.Status.CurrentReplicas)
+			blankResource.Labels = sset.Labels
+			blankResource.selector = sset.Spec.Selector.MatchLabels
+			rPool.addToResourcePool(&blankResource)
+		}
+	}
+	return rPool
+}
+
+func Int32toString(n int32) string {
```

Comment:
add comment

Reply:
and instead of this discrete function use `strconv.Itoa()`

Reply:
I read somewhere that the descrete function is the fastest way :p thats why I used it though

Reply:
If you are good and okay with it then I don't have any issue.

Reply:
It’s a temporary change. I think We should fetch status of children nodes and decide the status of parent node afterwards, so may not need this method